### ISSUE TITLE
🧪 [Experiment] MarkdownTextBlock

### ISSUE THREAD

**Issue by nerocui** [CommunityToolkit/Labs-Windows#606]
### Approved from Discussion

https://github.com/CommunityToolkit/Labs-Windows/discussions/536

### Problem Statement

Let's use this issue to track all improvements we are making to MardownTextBlock

### Overview

This experiment adds the following components:
 - **MarkdownTextBlock** a markdown control that uses the popular markdig parser for syntax parsing

### Using

You can try it out via the NuGet Packages here:
 - UWP: https://dev.azure.com/dotnet/CommunityToolkit/_artifacts/feed/CommunityToolkit-Labs/NuGet/CommunityToolkit.Labs.Uwp.Controls.MarkdownTextBlock/overview/0.1.241113-build.1784
 - WinUI 3: https://dev.azure.com/dotnet/CommunityToolkit/_artifacts/feed/CommunityToolkit-Labs/NuGet/CommunityToolkit.Labs.WinUI.Controls.MarkdownTextBlock/overview/0.1.241113-build.1784
Read more about [Preview Packages here](https://aka.ms/wct/wiki/previewpackages).

### Implementation Requirements

- [ ] Working Prototype
- [x] Feature Complete
- [x] Documentation
- [x] Samples
- [x] Unit Tests
- [x] Community Feedback / Usage Testimonies

### Tested Platforms

- [x] UWP
- [x] WinAppSDK / WinUI 3
- [ ] Web Assembly (WASM)
- [ ] Android
- [ ] iOS
- [ ] MacOS
- [ ] Linux / GTK

### Technical Review

- [ ] Accessibility Audit
- [ ] API/Naming Review
- [ ] Code Quality/Style
- [ ] Dependency Review
- [ ] Design/Style Review
- [ ] Final Approval

### Community Help?

Yes

**Comment by rstewa** [comment]
If you don't set the `MarkdownConfig` should it be set to `MarkdownConfig.Default` by default? Right now, if you don't set `MarkdownConfig` it doesn't render. If that's the expected behavior, then you can ignore this.

**Comment by nerocui** [comment]
> If you don't set the `MarkdownConfig` should it be set to `MarkdownConfig.Default` by default? Right now, if you don't set `MarkdownConfig` it doesn't render. If that's the expected behavior, then you can ignore this.

Yeah we'll address that in #611. Essentially no property should be "required" to just render some text.

**Comment by Sella-GH** [comment]
Hi, the `MarkdownTextBlock` apparently ignores the config setting `IsTextSelectionEnabled="False"`. I set this to `false` on every MTB I added in my app and I can still easily select the text.

Do I use this wrong, is there a bug or is it not implemented yet?

NuGet package: 0.1.260617-build.2640
.NET SDK: 10.0.400

<img width="621" height="195" alt="Image" src="https://github.com/user-attachments/assets/03594e52-6fd3-433a-8660-cec84d2ba317" />

<img width="691" height="257" alt="Image" src="https://github.com/user-attachments/assets/0e988090-e9cb-4c09-b3d1-bc18bf4be816" />
